Uploading pictures in comments

DoctorDib opened the pull request:
Changes:
 - Added image feature to comments
 - Added image preview component for comments
 - Fixed a small Image Uploader bug
 - Disabled ESLint on SnowPack DEV as it is annoyingly concating instead of refreshing the results which after a while will cause a slow down in console. (Run ESLint in a seperate console if you want it).
 - Small EsLint Changes

DoctorDib committed:
New image upload for comments

Changes:
 - Added image feature to comments
 - Added image preview component for comments
 - Fixed a small Image Uploader bug
 - Disabled ESLint on SnowPack DEV as it is annoyingly concating instead of refreshing the results which after a while will cause a slow down in console. (Run ESLint in a seperate console if you want it).
 - Small EsLint Changes

DoctorDib committed:
ESlint changes

Changes:
 - ESLint fixes
 - Switches from Img.onClick to Button.onClick

DoctorDib commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Ability to upload pictures to comments](https://trello.com/c/a7CQtVnl/79-ability-to-upload-pictures-to-comments)

BenRGH commented on client/snowpack.config.js:
```diff
@@ -14,11 +14,11 @@ module.exports = {
 
-    ["@snowpack/plugin-run-script", {
-      "cmd": "eslint \"src/**/*.{js,jsx,ts,tsx}\"",
+    /*["@snowpack/plugin-run-script", {
```
I disagree with this, you're moving the problem. At the very least we should have lint-staged in place before disabling.

BenRGH replied:
Also I haven't encountered this issue, the console shouldn't lag even with tens of thousands of lines

DoctorDib replied:
I've disabled it as it is causing problems, we'll just let everyone know that everyone must run "npm run format:eslint" for eslint tests or during development they can run "eslint watch". There is no point in keeping something that is causing us issues.

BenRGH replied:
we *have* to have linting before code is pushed to master, we can't trust everyone to run the script and remove code stinks

DoctorDib replied:
Then if they don't run / fix lints, their PR's will be rejected.

DoctorDib replied:
All we're doing is taking out ESLint checks on dev build, again it causes problems so no point it having it, unless you can find a fix

BenRGH replied:
Try getting lint-staged in, like I said I haven't had the issue. If you can't get lint-staged in then we'll need to drill it into the other guys to run the script.

DoctorDib replied:
I've uncommented it, I will only be disabling it whenever I am developing.

BenRGH commented on client/src/Components/Carousel/index.tsx:
```diff
@@ -13,21 +13,23 @@ import style from './styles';
-    const { owner, images } = props;
     const classes: IClasses = style();
 
+    const { owner, images, source } = props;
```
Props go at top

DoctorDib replied:
What's different between this file and other files that have the same structure

BenRGH replied:
The diff is I haven't noticed the others haha

BenRGH commented on client/src/Components/Comment/index.tsx:
```diff
@@ -50,6 +52,11 @@ const CommentComponent = React.memo((props: ICommentComponentProp): React.ReactE
     const [inputError, setInputError] = useState<boolean>(false);
     const [isExpanded, setExpand] = useState<boolean>(false);
+    const [images, setImages] = useState<Array<IImageSettings>>([]);
+
```
Unnecessary empty line

DoctorDib replied:
For you it's unnecessary, but for me it was OCD because the three useStates grouped together separately all have the same React.ReactElement type

BenRGH replied:
But they're all states so they should be grouped together, type shouldn't matter

DoctorDib replied:
It was been removed

BenRGH commented on client/src/Components/Comment/index.tsx:
```diff
@@ -130,6 +138,23 @@ const CommentComponent = React.memo((props: ICommentComponentProp): React.ReactE
+                            : (
+                                <section className={classes.quoteAttachment}>
+                                    <Typography variant="caption">
+                                        {' '}
```
We should really avoid these

DoctorDib replied:
We'll need to look into stopping ESLint from making those, it seems to be automatic

BenRGH replied:
Is it not prettier? Mine doesn't make them

DoctorDib replied:
It's 100% EsLint that does it, Prettier actually removes them.

It replaces spaces with {' '} but before those checks it put the contents of typography on a new line (with obviously spacing next before and after the contents, if that makes sense.

BenRGH replied:
have you tried not putting empty spaces around the contents

DoctorDib replied:
I mean <Typography> this </Typography> is usually what I write, it for some reason prefers it being on a different line unfortunately, causing those space problems.

DoctorDib replied:
I've removed {' '}

BenRGH commented on client/src/Components/Comment/index.tsx:
```diff
@@ -204,15 +240,21 @@ const CommentComponent = React.memo((props: ICommentComponentProp): React.ReactE
     };
 
+    const mapCommentImages = () => {
+        const commentImages = comment.post_attributes.comment_images;
```
type

BenRGH commented on client/src/Components/Comment/index.tsx:
```diff
@@ -204,15 +240,21 @@ const CommentComponent = React.memo((props: ICommentComponentProp): React.ReactE
         sendPost(threadID, '1', newPostAttributes, userType);
     };
 
+    const mapCommentImages = () => {
```
return type

BenRGH commented on client/src/Components/CommentTextBox/index.tsx:
```diff
@@ -25,14 +37,23 @@ const TextCommentComponent: React.FC<ICommentTextBoxProps> = (props: ICommentTex
     } = props;
 
+    const [openImageUploader, setOpenImageUploader] = useState<boolean>(false);
+    const [isExpanded, toggleExpand] = useState<boolean>(false);
```
better naming would be isExpanded, setIsExpanded. toggleExpand suggests it flips the value.

BenRGH replied:
setExpanded

BenRGH commented on client/src/Components/CommentTextBox/index.tsx:
```diff
@@ -51,18 +72,62 @@ const TextCommentComponent: React.FC<ICommentTextBoxProps> = (props: ICommentTex
+    const onClick = () => setOpenImageUploader(true);
+    const onClose = () => setOpenImageUploader(false);
+
+    const mapPreviewImages = () => tempImages.map((image:IImageSettings) => (
```
Return type

BenRGH commented on client/src/Components/Image/index.tsx:
```diff
@@ -0,0 +1,54 @@
+import { Dialog, IconButton, Button } from '@material-ui/core';
+
+import { Cancel } from '@material-ui/icons';
+
```
Wild line spaces here

BenRGH commented on client/src/Components/Image/index.tsx:
```diff
@@ -0,0 +1,54 @@
+    const [isFullScreen, setFullScreen] = useState<boolean>(false);
+
+    const onClose = () => setFullScreen(false);
+    const open = () => setFullScreen(true);
```
Needs better naming

BenRGH replied:
like onOpenClick

DoctorDib replied:
Which line

BenRGH replied:
22, its always the bottom line shown on github

DoctorDib replied:
Ahh, gotcha, will change it to onOpen

DoctorDib committed:
Requested PR Changes

Changes:
 - Added container background for comment images
 - ESLint fixes
 - Requested PR Changes

BenRGH commented on client/src/Components/Comment/index.tsx:
```diff
@@ -280,11 +277,15 @@ images attached
+                        ? ''
+                        : (
+                            <section className={classes.commentImageContainer}>
+                                {' '}
```
Spaces

DoctorDib replied:
Removed as requested

DoctorDib committed:
Small PR Change

Changes:
 - Removed {' '} created by ESLint

BenRGH commented on client/src/Components/Comment/index.tsx:
```diff
@@ -130,6 +137,21 @@ const CommentComponent = React.memo((props: ICommentComponentProp): React.ReactE
                     <section>
                         {formatAvatar(parentComment, classes, false)}
+                        {isNullOrUndefined(parentComment.post_attributes.comment_images)
+                            ? ''
```
Should probably return null instead of an empty string

DoctorDib replied:
Changes as requested

BenRGH commented on client/src/Components/Image/index.tsx:
```diff
@@ -0,0 +1,51 @@
+import React, { useState } from 'react';
+import { Dialog, IconButton, Button } from '@material-ui/core';
+import { Cancel } from '@material-ui/icons';
+import { IClasses } from '../../Common/Interfaces/IClasses';
```
line space between local and 3rd party

DoctorDib replied:
Changed as requested

BenRGH commented on client/src/Components/Posts/index.tsx:
```diff
@@ -32,6 +33,7 @@ const Forum = (props: IForumProps): React.ReactElement<IForumProps> => {
     const [commentValue, setCommentValue] = useState<string>('');
     const [inputError, setInputError] = useState<boolean>(false);
     const [comments, setComments] = useState<ReadonlyArray<React.ReactNode>>([]);
+    const [images, setImages] = useState<Array<IImageSettings>>([]);
```
ReadonlyArray

DoctorDib replied:
Changed as requested

DoctorDib committed:
Pr changes + small fix

Changes:
 - Fixed to clear images in on text box comment
 - Updated variable name to appropriate name.
 - Small Pr Changes
